### ISSUE TITLE
IgnoredReturnValue checking specific type in all functions

### ISSUE THREAD

**Issue by dkandalov** [detekt/detekt#6661]
The current `IgnoredReturnValue` implementation requires annotation on the function (https://github.com/detekt/detekt/blob/main/detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/IgnoredReturnValue.kt#L132C38-L132C60). It might be useful to make this optional, so you could specify a type which must always be handled.

For example, with `returnValueTypes: '*.Result'` the following code should be reported:
```
sealed class Result<out T, out E>
data class Success<out T>(val value: T) : Result<T, Nothing>()
data class Failure<out E>(val reason: E) : Result<Nothing, E>()

fun main() {
    Success(123)
}
```

**Comment by BraisGabin** [comment]
This has been implemented already. Closing.
